crates/protect-ffi/src/lib.rs

```diff
@@ -48,18 +83,26 @@ pub enum Error {
     Base85(String),
     #[error("unimplemented: {0} not supported yet by protect-ffi")]
     Unimplemented(String),
+    #[error(transparent)]
+    Parse(#[from] serde_json::Error),
+    #[error("column {}.{} not found in Encrypt config", _0.table, _0.column)]
+    UnknownColumn(Identifier),
 }
 
 type ScopedZeroKMSNoRefresh = ScopedCipher<ServiceCredentials>;
 
 fn new_client(mut cx: FunctionContext) -> JsResult<JsPromise> {
+    let encrypt_config_str = cx.argument::<JsString>(0)?.value(&mut cx);
+    let encrypt_config = EncryptConfig::from_str(&encrypt_config_str)
+        .or_else(|err| cx.throw_error(err.to_string()))?;
```

Comment:
The Encrypt config is passed in as a JSON-encoded string here and then parsed into the Rust struct. This looks like the best way to handle complex structs at the moment with Neon. This should get better once [extractors](https://github.com/neon-bindings/neon/pull/1024) are released.

There are several crates for integrating Serde with Neon ([1](https://crates.io/crates/neon-serde), [2](https://crates.io/crates/neon-serde2), [3](https://crates.io/crates/neon-serde3), [4](https://crates.io/crates/neon-serde4)), but the docs suggest that they don't support recent versions of Neon.

Reply:
Extractors look pretty good!